keras_cv/models/swin_transformer.py

```diff
@@ -0,0 +1,411 @@
+        patch_size=4,
+        window_size=12,
+        embed_dim=192,
+        depths=(2, 2, 18, 2),
+        num_heads=(6, 12, 24, 48),
+        name="swin_large_patch4_window12_384_in22k",
+        num_classes=21841,
+    )
+    return cfg
+
+
+def swin_large_patch4_window7_224_in22k():
+    """Swin-L @ 224x224, trained ImageNet-22k"""
+    cfg = dict(
+        patch_size=4,
+        window_size=7,
+        embed_dim=192,
+        depths=(2, 2, 18, 2),
+        num_heads=(6, 12, 24, 48),
+        name="swin_large_patch4_window7_224_in22k",
+        num_classes=21841,
+    )
+    return cfg
+
+
+def swin_s3_tiny_224():
+    """Swin-S3-T @ 224x224, ImageNet-1k. https://arxiv.org/abs/2111.14725"""
+    cfg = dict(
+        patch_size=4,
+        window_size=(7, 7, 14, 7),
+        embed_dim=96,
+        depths=(2, 2, 6, 2),
+        num_heads=(3, 6, 12, 24),
+        name="swin_s3_tiny_224",
+    )
+    return cfg
+
+
+def swin_s3_small_224():
+    """Swin-S3-S @ 224x224, trained ImageNet-1k. https://arxiv.org/abs/2111.14725"""
+    cfg = dict(
+        patch_size=4,
+        window_size=(14, 14, 14, 7),
+        embed_dim=96,
+        depths=(2, 2, 18, 2),
+        num_heads=(3, 6, 12, 24),
+        name="swin_s3_small_224",
+    )
+    return cfg
+
+
+def swin_s3_base_224():
+    """Swin-S3-B @ 224x224, trained ImageNet-1k. https://arxiv.org/abs/2111.14725"""
+    cfg = dict(
+        patch_size=4,
+        window_size=(7, 7, 14, 7),
+        embed_dim=96,
+        depths=(2, 2, 30, 2),
+        num_heads=(3, 6, 12, 24),
+        name="swin_s3_base_224",
+    )
+    return cfg
+
+
+MODEL_MAP = {
+    "swin_base_patch4_window12_384": swin_base_patch4_window12_384,
+    "swin_base_patch4_window7_224": swin_base_patch4_window7_224,
+    "swin_large_patch4_window12_384": swin_large_patch4_window12_384,
+    "swin_large_patch4_window7_224": swin_large_patch4_window7_224,
+    "swin_small_patch4_window7_224": swin_small_patch4_window7_224,
+    "swin_tiny_patch4_window7_224": swin_tiny_patch4_window7_224,
+    "swin_base_patch4_window12_384_in22k": swin_base_patch4_window12_384_in22k,
+    "swin_base_patch4_window7_224_in22k": swin_base_patch4_window7_224_in22k,
+    "swin_large_patch4_window12_384_in22k": swin_large_patch4_window12_384_in22k,
+    "swin_large_patch4_window7_224_in22k": swin_large_patch4_window7_224_in22k,
+    "swin_s3_tiny_224": swin_s3_tiny_224,
+    "swin_s3_small_224": swin_s3_small_224,
+    "swin_s3_base_224": swin_s3_base_224,
+}
+
+"""
+Code copied and modified from 
+https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/swin_transformer.py
+"""
+
+# https://github.com/rwightman/pytorch-image-models/blob/6d4665bb52390974e0cf9674c60c41946d2f4ee2/timm/models/layers/helpers.py#L10
+def to_ntuple(n):
+    def parse(x):
+        if isinstance(x, collections.abc.Iterable):
+            return x
+        return tuple(repeat(x, n))
+
+    return parse
+
+
+class SwinTransformer(keras.Model):
```

Comment:
Instead of subclassing, it's better to be built with functional.

cc. @LukeWood @sebastian-sz

Reply:
I think functional is more consistent with the majority of other models implementations in keras-cv? 

I don't have an opinion on which is better here.

Although I think it would be nice to have similar options as other implementations, like: extracting features could be achieved by setting `include_top` instead of using `forward_features`.

Reply:
Yup! That's why the ViT is built with a method using the functional API. This one was copy-pasted from Sayak's repo and is a placeholder for the functional port later. :)

Reply:
Naturally, once the port is finished - it'll conform as much as it can to the other networks in KerasCV, with as many of the shared arguments and commonalities in the API as there can be.
One thing that's worth noting is that we might not want to expose the ViT's features, since it has proven (afaik) itself to encode features that aren't very useful for most downstream tasks, and hence plain ViT's aren't used as backbones. Swins introduced hierarchical abstraction back into the mix, which made them a good backbone for downstream tasks :)

If we expose ViTs features, we could encourage people to use it as a backbone, if they aren't informed about the nuances between models?

Reply:
> If we expose ViTs features, we could encourage people to use it as a backbone, if they aren't informed about the nuances between models?

Yep!  This is a great point for using functional models.